diagnostic_tool/kinematics_cpp/kinematics.cpp

```diff
@@ -0,0 +1,443 @@
+/*
+ * Adapted from https://github.com/Le0nX/ModernRoboticsCpp

```

Comment:
1. I am not able to find a license for this - is that the original repo?
2. In this case, I would recommend create a PR that just copies the code, and then the modification on the original source code can live in a different PR. In this way, it is clear what has been changed compared to the original source code, and also it will be easier if we ever want to update the 3rd party dependency (though in this case ModernRobotics is unlikely to get any attention any more ... at least not from the original author...)

Reply:
Repo is a C++ implementation based off of https://github.com/NxRLab/ModernRobotics, which has MIT-license.